docs/src/index.md

```diff
@@ -0,0 +1,128 @@
+# LinearMaps.jl
+
+*A Julia package for defining and working with linear maps, also known as linear transformations or linear operators acting on vectors. The only requirement for a LinearMap is that it can act on a vector (by multiplication) efficiently.*
+
+## Installation
+
+`LinearMaps.jl` is a registered package and can be installed via
+
+    pkg> add LinearMaps
+
+in package mode, to be entered by typing `]` in the Julia REPL.
+
+## Examples
+
+Let
+
+    A = LinearMap(rand(10, 10))
+    B = LinearMap(cumsum, reverse∘cumsum∘reverse, 10)
+    
+be a matrix- and function-based linear map, respectively. Then the following code just works,
+indistinguishably from the case when `A` and `B` are both `AbstractMatrix`-typed objects.
+
+```
+3.0A + 2B
+A*B'
+[A B; B A]
+kron(A, B)
+```
+
+The `LinearMap` type and corresponding methods combine well with the following packages:
+* [Arpack.jl](https://github.com/JuliaLinearAlgebra/Arpack.jl): iterative eigensolver
+  `eigs` and SVD `svds`;
+* [IterativeSolvers.jl](https://github.com/JuliaMath/IterativeSolvers.jl): iterative
+  solvers, eigensolvers, and SVD;
+* [KrylovKit.jl](https://github.com/Jutho/KrylovKit.jl): Krylov-based algorithms for linear problems, singular value and eigenvalue problems
```

Comment:
So actually the compatibility with KrylovKit.jl could be increased if `LinearMap` instances could also be applied as a function, i.e. if there would be a single method
```julia
(L::LinearMap)(x::AbstractVector) = L*x
```

Reply:
Sounds good, makes sense.